### PR TITLE
repair: Invoke group0 read barrier in repair_tablets

### PR DESCRIPTION
**Pull request by asias**
This allows the repair master to see all previous metadata changes.

Refs #17658

### CODE REVIEW

**Review by denesb — APPROVED** [review]

File: repair/repair.cc [hunk 1]
```diff
@@ -2068,6 +2068,12 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
             throw std::runtime_error(format("repair[{}] Table {}.{} is not a tablet table", rid.uuid(), keyspace_name, table_name));
         }
         table_id tid = t->schema()->id();
+        // Invoke group0 read barrier before obtaining erm pointer so that it sees all prior metadata changes
+        auto dropped = co_await repair::table_sync_and_check(_db.local(), _mm, tid);
```
Comment (tgrabiec) [inline repair/repair.cc:2072]: This will invoke the barrier once for every table, since we iterate over table names sequentially.

**Review by tgrabiec — APPROVED** [review]

File: repair/repair.cc (hunk 1 above)
Comment (asias) [reply repair/repair.cc:2072]: Yes. I did this intentionally.  The metadata change might happen after we finish a table. Isn't better to do for each table than doing it only once.

Comment (tgrabiec) [reply repair/repair.cc:2072]: We need to ensure that effects of commands which finished before "nodetool repair" was started are observed by repair. There are no guarantees about effects of commands which started after "nodetool repair" was started. It's inherently racy. So I don't see a reason to repeat the barriers.

Comment (tgrabiec) [reply repair/repair.cc:2072]: Note that raft propagates group0 as fast as possible, executing a barrier doesn't make anything propagate faster. It's only for ensuring order of events.

**Pull request closed without merging**